File: mvvm/src/main/java/com/base/mvvm/feature/login/LoginActivity.java
package com.base.mvvm.feature.login;

import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.base.mvvm.R;
import com.base.mvvm.bean.UserBean;
import com.base.mvvm.util.PrefrenceUtil;

public class LoginActivity extends AppCompatActivity {
    private com.base.mvvm.databinding.ActivityLoginBinding binding;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_login);
        CheckUserInfoUtil.checkUserNameAndUserPwdIsInvalidate(binding);// RxJava联合判断
        binding.setClickHandler(new LoginHandler(binding));// 绑定监听
        UserBean userBean = PrefrenceUtil.getUserBean();
        if(userBean != null) {// 是否记住密码
            binding.cbIsRecordPwd.setChecked(userBean.isRecordPwd);
        } else {
            userBean = new UserBean();
        }
        binding.setUserBean(userBean);// 绑定用户信息
    }
}

File: kotlindemo/src/main/java/com/base/demo/login/LoginActivity.kt
package com.base.demo.login

import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.base.demo.R
import com.base.demo.bean.UserBean
import com.base.demo.databinding.ActivityLoginBinding

class LoginActivity : AppCompatActivity() {
    var loginBinding: ActivityLoginBinding ?= null
    var vm: LoginViewModel ?= null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        loginBinding = DataBindingUtil.setContentView(this, R.layout.activity_login)
        CheckUserInfoUtil.checkUserNameAndUserPwdIsInvalidate(loginBinding as ActivityLoginBinding)// RxJava联合判断
        vm = LoginViewModel(this)
        var userBean = UserBean()
        loginBinding?.userBean = userBean
        loginBinding?.loginHandler = LoginHandler(this)
    }
}
